### ISSUE TITLE
Wrap <csignal> include with ifdef

### ISSUE THREAD

**Issue by bnason-nf** [onqtam/doctest#177]
Hi @onqtam,

One of the platforms I am using doctest on doesn't support signals, so I made sure to disable DOCTEST_CONFIG_POSIX_SIGNALS. However, since this platform doesn't have the <csignal> header either, I also had to wrap that include with an ifdef:

```diff
diff --git a/doctest/doctest.h b/doctest/doctest.h
index eda93fc..9ebc1d0 100644
--- a/doctest/doctest.h
+++ b/doctest/doctest.h
@@ -2686,7 +2686,9 @@ DOCTEST_MAKE_STD_HEADERS_CLEAN_FROM_WARNINGS_ON_WALL_BEGIN
 #include <map>
 #include <exception>
 #include <stdexcept>
+#if defined(DOCTEST_CONFIG_POSIX_SIGNALS)
 #include <csignal>
+#endif
 #include <cfloat>
 #include <cctype>
 #include <cstdint>
```

I think it should be harmless if the platform does have signals. Is this something you would be willing to add to doctest? If so, please let me know if you'd like me to make a pull request.

Thanks,
Benbuck

**Comment by onqtam** [comment]
I just pushed the changes - the CI should pass the tests as well :)

**Comment by bnason-nf** [comment]
Thanks @onqtam. Sorry I forgot and provided the single include diff instead of the parts file.

-Benbuck

**Comment by cschreib** [comment]
This issue has reappeared in the current master branch. I noticed while trying to compile *doctest* to standalone wasm using Emscripten (standalone wasm doesn't have signals).

**Comment by AlexMax** [comment]
Can confirm, I am on a platform that does not have signals and I'm running into this error as well.

**Comment by mitchgrout** [comment]
Closed by #1054
